[BUG] `test_evaluate_for_global_mode` is failing

It seems that `test_evaluate_for_global_mode`, a test depending on `pytorch-forecasting`, is failing.
The exact failure mode is

```
Mismatched elements: 2 / 2 (100%)
Max absolute difference among violations: 86400000000000
 ACTUAL: array(['2000-01-05T00:00:00.000000000', '2000-01-05T00:00:00.000000000'],
      dtype='datetime64[ns]')
 DESIRED: array(['2000-01-04T00:00:00.000000000', '2000-01-04T00:00:00.000000000'],
      dtype='datetime64[ns]')
FAILED sktime/forecasting/model_evaluation/tests/test_evaluate.py::test_evaluate_global_mode[backend9-refit-scoring0] - AssertionError: 
Arrays are not equal
```

I.e., it seems like an incorrect index is being returned.

It would be good to investigate where the problem is coming from:

* the ptf adapter
* the splitter
* the base framework
* the `evaluate` method

@jgyasu , @phoeenniixx , any ideas?

--- a/sktime/forecasting/model_evaluation/_functions.py
+++ b/sktime/forecasting/model_evaluation/_functions.py
@@ -282,7 +282,16 @@ def _evaluate_window(x, meta):
                 temp_result[result_key] = [score]
 
         # get cutoff
-        cutoff = forecaster.cutoff
+        if global_mode:
+            # in global mode, cutoff should reflect y_hist (per-fold history),
+            # not the global y_train which may span more timepoints
+            if isinstance(y_hist.index, pd.MultiIndex):
+                last_time = y_hist.index.get_level_values(-1).max()
+            else:
+                last_time = y_hist.index[-1]
+            cutoff = pd.Index([last_time])
+        else:
+            cutoff = forecaster.cutoff
 
     except Exception as e:
         if error_score == "raise":

